### PR TITLE
[CI] remove redundant checks in `test_eager_matches_sdpa_inference`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Follow-up to #36650 

Removes redundant test cases in `test_eager_matches_sdpa_inference`, related to `output_attentions`.

SDPA has special attention mask preparation (e.g. see [here](https://github.com/huggingface/transformers/blob/e3af4fec91bdac3381e07f6e60fb8b74c3957c11/src/transformers/models/llama/modeling_llama.py#L657)) but, when we pass `output_attentions`, SDPA reverts to eager attention. That means that mask preparation for SDPA + `output_attentions` must be the same as eager. In #30652 we added a `for` loop to test the cases with `output_attentions in (True, False)`, resulting in 24 new subtests. 

That's testing overkill, all we need to check is to confirm that we're preparing the attention mask correctly with `output_attentions=True`. This PR removes all but 1 of the redundant tests.

@ydshieh with this PR, both CPU and GPU CI is faster than in `v4.49.0` 🤗

### CODE REVIEW

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -138,16 +138,16 @@
...
+        *(dtype, padding_side, use_attention_mask, False, enable_kernels),
     )
     for dtype in ("fp16", "fp32", "bf16")
     for padding_side in ("left", "right")
     for use_attention_mask in (True, False)
-    for output_attentions in (True, False)
     for enable_kernels in (True, False)
-]
+    # Extra test case: `output_attentions=True` has special attention mask handling and sdpa reverts to eager
+] + [("fp32_pad_left_output_attentions", "fp32", "left", True, True, False)]
```
Comment (gante) [inline tests/test_modeling_common.py:150]: padding-side left is the hardest test case: the causal mask will have 0 on the left (padding) and on the right (causal mask)

**Review by ydshieh — APPROVED** [review]
Very nice finding.

(BTW, would you like to add a reference to where shows this fact

> `output_attentions=True` has special attention mask

Might be helpful for further reader. (I have no idea, but you got my full trust. I am thinking for other people)

**Comment by gante** [comment]
Added a code link in the PR header for the special attention mask manipulation with SDPA + `output_attentions=True`, for future reference 👍
